javav2/example_code/rds/src/main/java/com/example/rds/RDSScenario.java

```diff
@@ -0,0 +1,526 @@
+import software.amazon.awssdk.services.rds.model.DeleteDbInstanceRequest;
+import software.amazon.awssdk.services.rds.model.DeleteDbInstanceResponse;
+import software.amazon.awssdk.services.rds.model.DescribeDbEngineVersionsRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbEngineVersionsResponse;
+import software.amazon.awssdk.services.rds.model.DescribeDbInstancesRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbInstancesResponse;
+import software.amazon.awssdk.services.rds.model.DescribeDbParameterGroupsResponse;
+import software.amazon.awssdk.services.rds.model.DescribeDbParametersResponse;
+import software.amazon.awssdk.services.rds.model.DescribeDbSnapshotsRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbSnapshotsResponse;
+import software.amazon.awssdk.services.rds.model.DescribeOrderableDbInstanceOptionsResponse;
+import software.amazon.awssdk.services.rds.model.ModifyDbParameterGroupResponse;
+import software.amazon.awssdk.services.rds.model.OrderableDBInstanceOption;
+import software.amazon.awssdk.services.rds.model.Parameter;
+import software.amazon.awssdk.services.rds.model.RdsException;
+import software.amazon.awssdk.services.rds.model.CreateDbParameterGroupRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbParameterGroupsRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbParametersRequest;
+import software.amazon.awssdk.services.rds.model.ModifyDbParameterGroupRequest;
+import software.amazon.awssdk.services.rds.model.DescribeOrderableDbInstanceOptionsRequest;
+import software.amazon.awssdk.services.rds.model.DeleteDbParameterGroupRequest;
+import java.util.ArrayList;
+import java.util.List;
+// snippet-end:[rds.java2.scenario.import]
+
+/**
+ * Before running this Java V2 code example, set up your development environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ *
+ * This Java example performs these tasks:
+ *
+ * 1. Returns a list of the available DB engines by invoking the DescribeDbEngineVersions method.
+ * 2. Select an engine family and create a custom DB parameter group by invoking the createDBParameterGroup method.
+ * 3. Get the parameter groups by invoking the DescribeDbParameterGroups methods.
+ * 4. Get parameters in the group by invoking the DescribeDbParameters method.
+ * 5. Modify both the auto_increment_offset parameters by invoking the modifyDbParameterGroup method.
+ * 6. Get and display the updated parameters.
+ * 7. Get a list of allowed engine versions by invoking the describeDbEngineVersions method.
+ * 8. Get a list of micro instance classes available for the selected engine
+ * 9. Create an RDS database instance that contains a MySql database and uses the parameter group
+ * 10. Wait for DB instance to be ready and print out the connection endpoint value.

```

Comment:
* 10. Waits for the DB instance to be ready and prints out the connection endpoint value.